src/vs/workbench/contrib/extensions/browser/extensionsViews.ts

```diff
@@ -650,6 +656,29 @@ export class ExtensionsListView extends ViewletPanel {
 			});
 	}
 
+	// {{SQL CARBON EDIT}}
+	private getRecommendedExtensionsByScenario(token: CancellationToken, scenarioType: string): Promise<IPagedModel<IExtension>> {
+		// if (!scenarioType) { return Promise.resolve(); }
```

Comment:
commented out? I think you can reject the promise in case of empty scenarioType.